source/common/config/well_known_names.cc

```diff
@@ -0,0 +1,86 @@
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Config {
+
+std::vector<std::pair<std::string, std::string>> TagNameValues::getRegexMapping() {
+  std::vector<std::pair<std::string, std::string>> name_regex_pairs;
+
+  // cluster.<cluster_name>.
+  name_regex_pairs.push_back({CLUSTER_NAME, "^cluster\\.((.*?)\\.)"});
+
+  // listener.<port>.
+  name_regex_pairs.push_back({LISTENER_PORT, "^listener\\.((\\d+?)\\.)"});
+
+  // http.<stat_prefix>.
+  name_regex_pairs.push_back({HTTP_CONN_MANAGER_PREFIX, "^http\\.((.*?)\\.)"});
+
+  // vhost.<virtual host name>.
+  name_regex_pairs.push_back({VIRTUAL_HOST, "^vhost\\.((.*?)\\.)"});
+
+  // mongo.<stat_prefix>.
+  name_regex_pairs.push_back({MONGO_PREFIX, "^mongo\\.((.*?)\\.)"});
+
+  // tcp.<stat_prefix>.
+  name_regex_pairs.push_back({TCP_PREFIX, "^tcp\\.((.*?)\\.)"});
+
+  // auth.clientssl.<stat_prefix>.
+  name_regex_pairs.push_back({CLIENTSSL_PREFIX, "^auth\\.clientssl\\.((.*?)\\.)"});
+
+  // ratelimit.<stat_prefix>.
+  name_regex_pairs.push_back({RATELIMIT_PREFIX, "^ratelimit\\.((.*?)\\.)"});
+
+  // http.<stat_prefix>.user_agent.<user_agent>.[base_stat]
+  name_regex_pairs.push_back({HTTP_USER_AGENT, "^http(?=\\.).*?\\.user_agent\\.((.*?)\\.)\\w+?$"});
+
+  // listener.<port>.ssl.cipher.<cipher>
+  name_regex_pairs.push_back({SSL_CIPHER, "^listener(?=\\.).*?\\.ssl\\.cipher(\\.(.*?))$"});
+
+  // mongo.<stat_prefix>.cmd.<cmd>.[base_stat]
+  name_regex_pairs.push_back({MONGO_CMD, "^mongo(?=\\.).*?\\.cmd\\.((.*?)\\.)\\w+?$"});
```

Comment:
So the use of lookahead here is to support both `mongo.foo.cmd` and `mongo.cmd` IIRC. I think you should change the comment to reflect this (and elsewhere). Also, The comment should indicate which capture groups there are, because `stat_prefix` isn't captured, only `cmd`. E.g. `mongo[.<stat_prefix>].cmd.(<cmd>).[base_stat]`.

Reply:
I added some notation to represent this along with an explanation of the notation at the top.